src/GeekLearning.Email/Internal/EmailSender.cs

```diff
@@ -96,12 +100,15 @@ private Task<ITemplate> GetTemplateAsync(string templateKey, EmailTemplateType t
             {
                 foreach (var recipient in recipients)
                 {
-                    var emailParts = recipient.Email.Split('@');
-                    if (emailParts.Length != 2)
+                
+                    string trimmedEmail = recipient.Email.Trim();
+                    // not sure if it's the most friendly to validate in the sender method. Should be left to caller code, IMO 
+                    if (Regex.IsMatch(trimmedEmail, @"\A(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)\Z", RegexOptions.IgnoreCase).Equals(false))
```

Comment:
I'm not comfortable with regex especially without a timeout in our codebase. I'm not sure this is the role of the library to validate input altogether.  So I'm all in for dropping it entirely. @asiffermann you may have a different opinion.

Reply:
Yes. And there was a throw in the original implementation when checking the domain part to conditional branch into the mocking sender insertion. Is there a way to check SMTP errors in the SmtpClient and then throw for failures there? Then at least the user code takes responsibility for address validation, but if that isn't done, at least the app will get alerted at the real point of failure.